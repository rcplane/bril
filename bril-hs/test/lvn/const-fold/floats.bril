@main {
  feight: float = const 8.0;
  ftwo: float = const 2.0;

  fsum: float = fadd feight ftwo;
  fprod: float = fmul feight ftwo;
  fdiff: float = fsub feight ftwo;
  fquot: float = fdiv feight ftwo;

  print fsum fprod fdiff fquot;

  feq: bool = feq feight ftwo;
  flt: bool = flt feight ftwo;
  fgt: bool = fgt feight ftwo;
  fle: bool = fle feight ftwo;
  fge: bool = fge feight ftwo;

  print feq flt fgt fle fge;
}
