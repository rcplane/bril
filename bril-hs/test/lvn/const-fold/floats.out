@main {
  fsum: float = const 10;
  fprod: float = const 16;
  fdiff: float = const 6;
  fquot: float = const 4;
  print fsum fprod fdiff fquot;
  feq: bool = const false;
  flt: bool = const false;
  fgt: bool = const true;
  fle: bool = const false;
  fge: bool = const true;
  print feq flt fgt fle fge;
}
